src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerAssemblyLoader.cs

```diff
@@ -424,5 +427,41 @@ internal string GetRealAnalyzerLoadPath(string originalFullPath)
             }
             return null;
         }
+
+        internal string? RedirectAssemblyPathExternally(string fullPath)
+        {
+            CheckIfDisposed();
+
+            if (!_externalRedirectors.IsDefaultOrEmpty)
+            {
+                string? redirectedPath = null;
+
+                foreach (IAnalyzerAssemblyRedirector redirector in _externalRedirectors)
+                {
+                    try
+                    {
+                        if (redirector.RedirectPath(fullPath) is { } currentlyRedirectedPath)
+                        {
+                            if (redirectedPath == null)
+                            {
+                                redirectedPath = currentlyRedirectedPath;
+                            }
+                            else if (redirectedPath != currentlyRedirectedPath)
```

Comment:
Given this is a path lets use `string.Equals` with an explicit comparer.

Reply:
`==` is equivalent to ordinal comparison; so did you mean to use ignore-case ordinal comparison, or just be more explicit?